tests/integrationv2/test_session_resumption.py

```diff
@@ -320,3 +320,72 @@ def test_s2nd_falls_back_to_full_connection(managed_process, tmp_path, cipher, c
         assert b'Resumed session' not in results.stdout
         assert to_bytes("Actual protocol version: {}".format(
             s2n_version)) in results.stdout
+
+
+@pytest.mark.uncollect_if(func=invalid_test_parameters)
+@pytest.mark.parametrize("cipher", ALL_TEST_CIPHERS, ids=get_parameter_name)
+@pytest.mark.parametrize("curve", ALL_TEST_CURVES, ids=get_parameter_name)
+@pytest.mark.parametrize("certificate", ALL_TEST_CERTS, ids=get_parameter_name)
+@pytest.mark.parametrize("protocol", [p for p in PROTOCOLS if p < Protocols.TLS13], ids=get_parameter_name)
+@pytest.mark.parametrize("provider", [OpenSSL, S2N], ids=get_parameter_name)
+@pytest.mark.parametrize("other_provider", [S2N], ids=get_parameter_name)
+def test_session_resumption_s2n_client_tls13_server_not_tls13(managed_process, cipher, curve, protocol, provider, other_provider, certificate):
```

Comment:
Nit: you shouldn't actually need "other_provider" anymore if it's just S2N. There was a bug in invalid_test_parameters that required it, but I fixed that a while back.